src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,29 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+export const TodoModal: React.FC<{
+  selectedTodo: Todo;
+  setSelectedTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+}> = ({ selectedTodo: { completed, id, title, userId }, setSelectedTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    getUser(userId).then(userFromServer => {
+      setUser(userFromServer);
+      setIsLoading(false);
+    });
+  }, []);
```

Comment:
don't you have a missing dependency ?

Reply:
What should it be dependent on? We only need the user once, since that component gets unmounted and remounted for the next user, if that's what you mean?